spec/index.bs

```diff
@@ -2282,6 +2282,46 @@ dictionary GPUImageBitmapCopyView {
 
   * {{GPUImageBitmapCopyView/origin}}: If unspecified, defaults to `[0, 0]`.
 
+### <dfn method for=GPUCommandEncoder>copyBufferToBuffer(source, sourceOffset, destination, destinationOffset, size)</dfn> ### {#GPUCommandEncoder-copyBufferToBuffer}
+
+<div algorithm="GPUCommandEncoder.copyBufferToBuffer">
+
+  **Arguments:**
+    - {{GPUBuffer}} |source|
+    - {{GPUSize64}} |sourceOffset|
+    - {{GPUBuffer}} |destination|
+    - {{GPUSize64}} |destinationOffset|
+    - {{GPUSize64}} |size|
+
+  **Returns:** void
+
+  Encode a command into the {{GPUCommandEncoder}} that copies |size| bytes of data from the |sourceOffset| of a {{GPUBuffer}} |source| to the |destinationOffset| of another {{GPUBuffer}} |destination|.
+
+</div>
+
+<div algorithm class=validusage>
+
+  <dfn abstract-op>copyBufferToBuffer Valid Usage</dfn>
+
+  Given a {{GPUCommandEncoder}} |encoder| and the arguments {{GPUBuffer}} |source|, {{GPUSize64}} |sourceOffset|, {{GPUBuffer}} |destination|, {{GPUSize64}} |destinationOffset|, {{GPUSize64}} |size|, the following validation rules apply:
+
+  - |encoder|.{{GPUCommandEncoder/copyBufferToBuffer()}} must be called before calling |encoder|.{{GPUCommandEncoder/finish()}}. // TODO: define the state machine for GPUCommandEncoder.
+  - |encoder|.{{GPUCommandEncoder/copyBufferToBuffer()}} must be called outside of any {{GPURenderPassEncoder}}s.
+  - |encoder|.{{GPUCommandEncoder/copyBufferToBuffer()}} must be called outside of any {{GPUComputePassEncoder}}s.
+  - |source| must be a [=valid=] {{GPUBuffer}}.
+  - |destination| must be a [=valid=] {{GPUBuffer}}.
+  - The {{GPUBuffer/[[usage]]}} of |source| must contain {{GPUBufferUsage/COPY_SRC}}.
+  - The {{GPUBuffer/[[usage]]}} of |destination| must contain {{GPUBufferUsage/COPY_DST}}.
+  - |size| must be a multiple of 4.
+  - |sourceOffset| must be a multiple of 4.
+  - |destinationOffset| must be a multiple of 4.
+  - (|sourceOffset| + |size|) must not overflow a {{GPUBuffer/[[size]]}}. // TODO: figure out how to handle overflows in the spec.
```

Comment:
isn't it the same as " The {{GPUBuffer/[[size]]}} of |source| must be greater than or equal to (|sourceOffset| + |size|)." that you have later in the list?

Reply:
I think it was supposed to say "GPUBufferSize": https://github.com/gpuweb/gpuweb/pull/597#discussion_r389577917

But now that you mention it I think that's also redundant. If it's <= the buffer's size, then it must be <= max GPUBufferSize.

This is fine as long as "+" is treated as a mathematical operator on real numbers/integers (I think it should be), not a CPU instruction on int32s/int64s.

Reply:
Oh {{GPUBufferSize}} has not been defined yet, do you mean define it in this PR?

Reply:
Oops, I'm sorry, this got renamed to `{{GPUSize64}}` and we forgot.

Reply:
Done